joblib/test/test_parallel.py

```diff
@@ -151,19 +157,74 @@ def test_main_thread_renamed_no_warning(backend, monkeypatch):
         results = Parallel(n_jobs=2, backend=backend)(
             delayed(square)(x) for x in range(3))
         assert results == [0, 1, 4]
+
+    # Due to the default parameters of LokyBackend, there is a chance that
+    # warninfo catches Warnings from worker timeouts. We remove it if it exists
+    warninfo = [w for w in warninfo if "worker timeout" not in str(w.message)]
+
     # The multiprocessing backend will raise a warning when detecting that is
     # started from the non-main thread. Let's check that there is no false
     # positive because of the name change.
     assert len(warninfo) == 0
 
 
+def _assert_warning_nested(backend, inner_n_jobs, expected):
+    # Makes sure the warnings are always thrown
+    warnings.resetwarnings()
```

Comment:
Not sure whether these two lines are really needed. If they are I think it is better to move them inside the `with warns(None)` block in order not to modify the global warnings settings.